compiler/rustc_resolve/src/errors.rs

```diff
@@ -926,6 +910,27 @@ pub(crate) struct UnnamedCrateRootImport {
     pub(crate) span: Span,
 }
 
+#[derive(Subdiagnostic)]
+#[multipart_suggestion(
+    resolve_unnamed_super_or_self_sugg_import,
+    applicability = "maybe-incorrect",
+    style = "verbose"
+)]
+pub(crate) struct UnnamedSuperOrSelfImportSugg {
+    #[suggestion_part(code = "{ident} as name")]
+    pub(crate) span: Span,
+    pub(crate) ident: Ident,
+}
+
+#[derive(Diagnostic)]
+#[diag(resolve_unnamed_super_or_self_import)]
+pub(crate) struct UnnamedSuperOrSelfImport {
```

Comment:
```suggestion
pub(crate) struct UnnamedImport {
```
And in other errors too.
This is more general than just about super and self.